### PR TITLE
RHTAP-1134: Build service mount RH certs

### PR DESCRIPTION
**Pull request by gbenhaim**
Required for accessing RH Gitlab instance.

### CODE REVIEW

**Review by mmorhun — COMMENTED** [review]
I think, this should be done on cluster level.
The certificates should be trusted by OpenShift, then every pod (including Build Service) would trust them.

File: components/build-service/components/rh-certs/2015-RH-IT-Root-CA.pem [hunk 1]
```diff
@@ -0,0 +1,25 @@
+-----BEGIN CERTIFICATE-----
```
[the remark this answers is not in the document]
Comment (gbenhaim) [reply components/build-service/components/rh-certs/2015-RH-IT-Root-CA.pem:1]: no, since they are not secrets.

**Comment by gbenhaim** [comment]
@mmorhun I didn't find a way to add a cluster wide CA for user workloads (I searched in the OCP docs). There is only a way to add a CA to the infrastructure components. Do you maybe know how to do configure it for user workloads?

File: components/build-service/components/rh-certs/2015-RH-IT-Root-CA.pem (hunk 1 above)
[the remark this answers is not in the document]
Comment (gbenhaim) [reply components/build-service/components/rh-certs/2015-RH-IT-Root-CA.pem:1]: @scoheb can I close this thread?

[the remark this answers is not in the document]
Comment (chmeliik) [reply components/build-service/components/rh-certs/2015-RH-IT-Root-CA.pem:1]: Is it really OK to expose Red Hat certificates like this? Are they supposed to be publically available?

File: .kube-linter.yaml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+checks:
+  ignorePaths:
+    - components/build-service/components/rh-certs/add-rh-certs-patch.yaml
```
Comment (chmeliik) [inline .kube-linter.yaml:3]: How does the add-certs patch fail linting?

**Review by chmeliik — COMMENTED** [review]
Is it intentional that the certs are only applied to the dev env and one stage env?
* components/build-service/development/kustomization.yaml
* components/build-service/staging/stone-stage-p01/kustomization.yaml

**Comment by gbenhaim** [comment]
@mmorhun the proxy affects only the system components:
```
System-wide proxy affects system components only, not user workloads
```
https://docs.openshift.com/container-platform/4.14/networking/enable-cluster-wide-proxy.html#prerequisites

File: .kube-linter.yaml (hunk 2 above)
Comment (gbenhaim) [reply .kube-linter.yaml:3]: because it doesn't contain a full definition of a deployment (and this is ok)

**Comment by gbenhaim** [comment]
| Is it intentional that the certs are only applied to the dev env and one stage env?

components/build-service/development/kustomization.yaml
components/build-service/staging/stone-stage-p01/kustomization.yaml

@chmeliik yes, since those are the only instances I expect will require those public certs

File: components/build-service/components/rh-certs/2015-RH-IT-Root-CA.pem (hunk 1 above)
[the remark this answers is not in the document]
Comment (gbenhaim) [reply components/build-service/components/rh-certs/2015-RH-IT-Root-CA.pem:1]: @chmeliik I don't think it's an issue. They don't contain any secret data, just a public key.

File: .kube-linter.yaml (hunk 2 above)
Comment (gbenhaim) [reply .kube-linter.yaml:3]: ok, cool. I ran kube-linter on the entire directory (not just on the output of kustomize build). I'll remove the config file.

Comment (gbenhaim) [reply .kube-linter.yaml]: done

**Comment by gbenhaim** [comment]
any other comments? I would like to get it merged today.

File: components/build-service/components/rh-certs/2015-RH-IT-Root-CA.pem (hunk 1 above)
[the remark this answers is not in the document]
Comment (chmeliik) [reply components/build-service/components/rh-certs/2015-RH-IT-Root-CA.pem:1]: Ok, seems reasonable

**Comment by chmeliik** [comment]
> | Is it intentional that the certs are only applied to the dev env and one stage env?
> 
> components/build-service/development/kustomization.yaml components/build-service/staging/stone-stage-p01/kustomization.yaml
> 
> @chmeliik yes, since those are the only instances I expect will require those public certs

How come prod doesn't need them? Could you add the explanation to the commit message?

**Review by chmeliik — APPROVED** [review]
LGTM pending explanation for why the certs aren't needed in prod

**Review by mmorhun — APPROVED** [review]

**Pull request merged**